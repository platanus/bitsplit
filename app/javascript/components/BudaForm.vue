<template>
  <div class="mt-16">
    <center>
      <div class="w-full max-w-xs">
        <form @submit.prevent="handleSubmit">
          <div>
            <textField>
              Ingresa el API_KEY y API_SECRET de tu cuenta Buda. Puedes acceder a ellos en tu perfil de <a href="https://www.buda.com">www.buda.com</a>
            </textField>
          </div>
          <div>
            <inputLabel field-name="API_KEY">
              key
            </inputLabel>
            <passwordInput
              field-id="apiKey"
              field-name="apiKey"
              v-model="apiKey"
            />
          </div>
          <div>
            <inputLabel field-name="API_SECRET">
              apiSecret
            </inputLabel>
            <passwordInput
              field-id="apiSecret"
              field-name="apiSecret"
              v-model="apiSecret"
            />
          </div>
          <div>
            <inputLabel field-for="password">
              Confirma tu contrase√±a Bitsplit
            </inputLabel>
            <passwordInput
              field-id="password"
              field-name="password"
              v-model="password"
            />
          </div>
          <div>
            <submitButton :loading="budaSignInLoading">
              Sincronizar Buda
            </submitButton>
          </div>
        </form>
      </div>
    </center>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex';
import passwordInput from '../components/PasswordInput';
import submitButton from '../components/SubmitButton';
import inputLabel from '../components/InputLabel';
import textField from '../components/TextField';

export default {
  name: 'BudaForm',
  data() {
    return {
      routeName: 'BudaForm',
      apiKey: '',
      apiSecret: '',
      password: '',
    };
  },
  components: {
    passwordInput,
    submitButton,
    inputLabel,
    textField,
  },
  computed: {
    ...mapState('user', ['currentUser', 'budaSignInLoading']),
  },
  methods: {
    ...mapActions('user', ['budaSignIn']),
    handleSubmit() {
      const { apiKey, apiSecret, password } = this;
      if (apiKey && apiSecret && password) {
        this.budaSignIn({
          api_key: apiKey,
          api_secret: apiSecret,
          password,
        })
          .then(() => {
            this.$router.push('/home');
          })
          .catch(err => {
            console.error(err);
          });
      }
    },
  },
};
</script>
