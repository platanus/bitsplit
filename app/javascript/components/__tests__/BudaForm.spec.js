import { shallowMount } from '@vue/test-utils'
import BudaForm from '../BudaForm'

describe('BudaForm', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(BudaForm)
    expect(wrapper).toMatchSnapshot()
  })
})
