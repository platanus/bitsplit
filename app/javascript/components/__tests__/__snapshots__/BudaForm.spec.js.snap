// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudaForm matches snapshot 1`] = `
<div id="budaSignInForm">
  <center>
    <div class="w-full max-w-xs">
      <form>
        <div>
          <textfield-stub textvalue1="Ingresa el API_KEY y API_SECRET de tu cuenta Buda. Puedes acceder a ellos en tu perfil de" linktext="www.buda.com" href="https://www.buda.com"></textfield-stub>
        </div>
        <div>
          <inputlabel-stub fieldname="API_KEY" fieldfor="key"></inputlabel-stub>
          <passwordinput-stub value="" fieldid="api_key" fieldname="api_key"></passwordinput-stub>
        </div>
        <div>
          <inputlabel-stub fieldname="API_SECRET" fieldfor="api_secret"></inputlabel-stub>
          <passwordinput-stub value="" fieldid="api_secret" fieldname="api_secret"></passwordinput-stub>
        </div>
        <div>
          <inputlabel-stub fieldname="Confirma tu contraseÃ±a Bitsplit" fieldfor="password"></inputlabel-stub>
          <passwordinput-stub value="" fieldid="password" fieldname="password"></passwordinput-stub>
        </div>
        <div>
          <submitbutton-stub fieldplaceholder="Sincronizar Buda"></submitbutton-stub>
        </div>
      </form>
    </div>
  </center>
</div>
`;
